ktor-client/ktor-client-android/jvm/src/io/ktor/client/engine/android/AndroidClientEngine.kt

```diff
@@ -62,7 +62,7 @@ public class AndroidClientEngine(override val config: AndroidEngineConfig) : Htt
             config.requestConfig(this)
 
             if (data.method in METHODS_WITHOUT_BODY) {
-                if (outgoingContent is OutgoingContent.NoContent) {
+                if (outgoingContent is OutgoingContent.NoContent || data.isSseRequestWithoutBody()) {
```

Comment:
Seems like an abstraction reversal here with the SSE checks... It might make sense to introduce a `OutgoingContent.ContentWrapper` type that just delegates to an `OutgoingContent` property, then add an `isEmpty()` function to `OutgoingContent` that returns false for everything except `NoContent` and the new wrapper type which delegates.  Then this would just be `if (outgoingContent.isEmpty())`.

Reply:
Thanks a lot for the refactoring suggestions! I think it makes sense to introduce `OutgoingContent.ContentWrapper`. I wasn't really sure about `ResponseAdapter` in the comment lower because it looks like `SseResponseAdapter` is the only usage of it, but please correct me if I am wrong

Reply:
Oh yeah, it's more about providing an extension point for future similar plugins - seems to be a common enough pattern anyway.